File: 09-week-7--sorting-algorithms/00-days-1-to-2--bubble-sort/javascript/bubble_sort.js
function bubbleSort(arr) {
  // type your code here

  for (let j = 0; j < arr.length; j++){

    for (let i = 0; i < arr.length - i; i++) {
      if (arr[i] > arr[i+1]) {
        let second = arr[i+1];
        arr[i + 1] = arr[i];
        arr[i] = second;
      }
    }
  
  }

  return arr;
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: [1, 2, 3, 4]");
  console.log("=>", bubbleSort([3, 2, 1, 4]));

  console.log("");

  console.log("Expecting: [1, 2, 3]");
  console.log("=>", bubbleSort([1, 2, 3]));

  console.log("");

  console.log("Expecting: []");
  console.log("=>", bubbleSort([]));

  console.log("");

  console.log("Expecting: [1, 2, 3]");
  console.log("=>", bubbleSort([2, 3, 1]));
}

module.exports = bubbleSort;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: 09-week-7--sorting-algorithms/00-days-1-to-2--bubble-sort/ruby/bubble_sort.rb
def bubble_sort(arr)
  # type your code in here
  sorted = false

  while(!sorted)
    sorted = true

    arr.each_index do |idx|
      break if idx == arr.length - 1
      
      if arr[idx] > arr[idx + 1]
        second = arr[idx + 1]
        arr[idx + 1] = arr[idx]
        arr[idx] = second
        sorted = false
      end

    end
  end

  arr
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: [1, 2, 3, 4]"
  print "=> "
  print bubble_sort([3, 2, 1, 4])

  puts

  puts "Expecting: [1, 2, 3]"
  print "=> "
  print bubble_sort([1, 2, 3])

  puts

  puts "Expecting: []"
  print "=> "
  print bubble_sort([])

  puts

  puts "Expecting: [1, 2, 3]"
  print "=> "
  print bubble_sort([2, 3, 1])

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
